x/finality/types/finality.go

```diff
@@ -9,8 +9,52 @@ import (
 
 	"github.com/babylonlabs-io/babylon/v3/crypto/eots"
 	"github.com/babylonlabs-io/babylon/v3/types"
+	btcstktypes "github.com/babylonlabs-io/babylon/v3/x/btcstaking/types"
 )
 
+// FinalityProviderState defines the possible states of a finality provider
+// It is used in the power distribution change process
+// to determine the state of each finality provider and how it affects the power distribution
+type FinalityProviderState int32
+
+const (
+	// FinalityProviderState_UNKNOWN indicates the finality provider state is unknown or uninitialized
+	FinalityProviderState_UNKNOWN FinalityProviderState = iota
+	// FinalityProviderState_UNJAILED indicates the finality provider is active and can participate
+	FinalityProviderState_UNJAILED
+	// FinalityProviderState_JAILED indicates the finality provider is jailed and cannot participate
+	FinalityProviderState_JAILED
+	// FinalityProviderState_SLASHED indicates the finality provider has been slashed
+	FinalityProviderState_SLASHED
+)
+
+// Processing state during the power distribution change process
+// It holds the state of finality providers, BTC delegations, and events
+// It is used to track the changes in the finality providers' states and the BTC delegations
+// It is also used to determine the final state of each finality provider after the power distribution change
+// The state is updated during the power distribution change process and is used to generate the
+// final power distribution cache
+type ProcessingState struct {
+	// FPStatesByBtcPk is a map of the finality providers' state
+	FPStatesByBtcPk map[string]FinalityProviderState
+	FpByBtcPk       map[string]*btcstktypes.FinalityProvider
+	// DeltaSatsByFpBtcPk is a map where key is finality provider's BTC PK hex and value is the
+	// delta of BTC delegations satoshis that were added or removed from the provider
+	// during the power distribution change process
+	DeltaSatsByFpBtcPk map[string]int64
```

Comment:
int64 range is -9223372036854775808 through 9223372036854775807
considering 1 sat is 0.00000001 BTC, and there'll be only 21 M BTC, we should be OK,
as the delta int64 range in BTC would be -92,233,720,368.54775808 thru 92,233,720,368.54775808